hack/test-integration.sh

```diff
@@ -69,6 +69,7 @@ os::log::start_system_logger
 function exectest() {
 	echo "Running $1..."
 
+	export TEST_ETCD_DIR=$(mktemp ${TMPDIR:-/tmp}/etcd-XXXXXXXX)
```

Comment:
we need a clean etcd dir per test and the ability to retain etcd data for (possibly multiple) failed tests, which means we can't have a single etcd dir